client/systems.go

```diff
@@ -186,12 +186,21 @@ type SystemDetails struct {
 	StorageEncryption *StorageEncryption `json:"storage-encryption,omitempty"`
 }
 
+func setEnclosingVolumeInStructs(vv map[string]*gadget.Volume) {
+	for _, v := range vv {
+		for sidx := range v.Structure {
+			v.Structure[sidx].EnclosingVolume = v
+		}
+	}
+}
+
```

Comment:
feels like this should be a helper in gadget proper? and remove the one in gadgettest?